src/osdc/ObjectCacher.cc

```diff
@@ -1778,6 +1778,11 @@ int ObjectCacher::writex(OSDWrite *wr, ObjectSet *oset, Context *onfreespace,
     bh->last_write = now;
 
     o->try_merge_bh(bh);
+    if (o->data.size() >= cct->_conf->client_oc_max_object_ops
```

Comment:
This should really be testing the number of dirty `bh`s correct? Otherwise you will keep calling flush over and over again for potentially clean objects.

Reply:
I'm sorry, I don't quite understand what you mean.Isn't it all dirty data of the object in o->data?

Reply:
See the `flush` method -- `o->data` is a map of offsets to `BufferHead`s. Each `BufferHead` has a state associated w/ it (i.e. clean, dirty, rx, tx, ...). You should only care about the dirty `BufferHead`s since those are the only ones that would get flushed out to the OSDs.

Reply:
... for efficiency sake, you could also split up the current `Object::dirty_or_tx` into `Object::num_dirty` and `Object::num_tx` so that you don't need to perform a potentially expensive O(n) search through all the bhs to count them up for each write op.